upgrade.sh

```diff
@@ -114,11 +176,28 @@ main() {
   # Replace `build-snaps` for `engine` with $GO_VERSION
   yq_p '.parts.engine."build-snaps"[0] |= sub("[0-9]+\.[0-9]+", "'"$GO_VERSION"'")' "$yaml_file"
 
-  # Replace the remaining comments
-  sed -i "s/moby\/blob\/$CURRENT/moby\/blob\/$LATEST/g" "$yaml_file"
-
   echo "YAML file updated successfully."
+}
+
+cleanup_vm() {
+  echo "Cleaning up VM..."
+  multipass delete "${VM_NAME}"
+  multipass purge
```

Comment:
purge removes data of all deleted instances, more than what this script creates. Instead, make the VM name unique, e.g. by adding a date and not purge.

Reply:
Not freeing up vs freeing up things that shouldn't be freed up.

Maybe this can be a feature request for multipass.